### ISSUE TITLE
v2.0.0 now places the party in the upper left-hand corner

### ISSUE THREAD

**Issue by peterbe** [yiliansource/party-js#60]
**Describe the bug**
The sparkles (or confetti) are now in the upper left-hand corner of the page. 

<img width="329" alt="Screen Shot 2021-05-13 at 8 19 52 AM" src="https://user-images.githubusercontent.com/26739/118124887-350df080-b3c4-11eb-94b9-6c3543c45ede.png">

Here's how I run it:
```jsx
<input
  onClick={(event) => {
    event.stopPropagation();
    if (true) {
      party.sparkles(event.target as HTMLElement, {
      });
    }
  }}
/>{
```

**To Reproduce**
1. Go to https://codesandbox.io/s/cool-paper-9urhf?file=/src/App.tsx

**Expected behavior**

**Enviroment**
- [x] Browser
- [ ] Node.js
- [ ] Other

**Additional context**

v2.0.0 just came out.

**Comment by yiliansource** [comment]
And I had hoped that I'd have ironed out all the quirky bugs at this point...

Thanks for the reproduceable example, I'll try to find a fix tomorrow when I'm back at my desktop. When testing this morning it worked fine on both bundle and package version.

**Comment by peterbe** [comment]
Is my Codesandbox sane though? 
That one uses React + Typescript. My other project, where I first discovered this, was Preact + Typescript but I don't know if that should matter. 

Thanks in advance! It's a great looking lib!

**Comment by yiliansource** [comment]
I'm not even sure what's sane anymore. 😅

The issue seems to be that the renderer isn't applying the position to the particles, but it's not throwing errors. Might be generating faulty CSS declarations, but I'll have to look into it.

**Comment by yiliansource** [comment]
Apparently the particle instances have their `location` set to `(NaN, NaN, 0)`, which causes the CSS styles to become corrupted. I'll see if I can make out the reason why.

**Comment by yiliansource** [comment]
Alright, I figured it out!

The `party.dynamicSource(source)` method tries to infer the source to use from the passed element by checking the available properties. For normal `<div>` elements this worked fine, but `<input>` elements have an additional `width` property, which makes the dynamic source think that the source is a Rect, not an HTMLElement.

The workaround is to use `party.elementSource(source)`, but I'll push a fix later today probably.

**Comment by yiliansource** [comment]
Fixed in v2.0.1 by using `instanceof` guards instead of `in` clauses.

**Comment by yiliansource** [comment]
Fixed in v2.0.1 by using `instanceof` guards instead of `in` clauses.

**Comment by peterbe** [comment]
Forked. Upgraded. Confirmed to fixed: https://codesandbox.io/s/gracious-bush-pj564?file=/src/App.tsx
